ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml

```diff
@@ -269,39 +282,75 @@ SettingsPageLayout {
 
-                    onPreviewClicked: console.log("TODO: Assets preview!")
+                    onPreviewClicked: {
+                        d.accountAddress = accountAddress
+                        stackManager.push(d.previewTokenViewState,
+                                          previewTokenView,
+                                          {
+                                              preview: true,
+                                              isAssetView: true,
+                                              name,
+                                              artworkSource,
+                                              artworkCropRect,
+                                              symbol,
+                                              description,
+                                              supplyAmount,
+                                              infiniteSupply,
+                                              assetDecimals,
+                                              chainId,
+                                              chainName,
+                                              chainIcon,
+                                              accountName
+                                          },
+                                          StackView.Immediate)
+                    }
                 }
             }
         }
     }
 
     Component {
-        id: previewCollectibleView
+        id: previewTokenView
 
-        CommunityCollectibleView {
+        CommunityTokenView {
             id: preview
 
             function signMintTransaction() {
                 root.setFeeLoading()
-                root.mintCollectible(artworkSource,
-                                     name,
-                                     symbol,
-                                     description,
-                                     supplyAmount,
-                                     infiniteSupply,
-                                     transferable,
-                                     selfDestruct,
-                                     chainId,
-                                     accountName,
-                                     d.accountAddress,
-                                     artworkCropRect)
+                if(preview.isAssetView) {
+                    root.mintAsset(artworkSource,
+                                   name,
+                                   symbol,
+                                   description,
+                                   supplyAmount,
+                                   infiniteSupply,
+                                   assetDecimals,
+                                   chainId,
+                                   accountName,
+                                   d.accountAddress,
+                                   artworkCropRect)
+                } else {
+                    root.mintCollectible(artworkSource,
+                                         name,
```

Comment:
Would it make sense to define an `Asset` object. I see these are the properties that define an asset and it's used in multiple places

Reply:
Yeah..  I will think on that. I still have other pending tasks to work on that feature. I will see if I can do it more elegante! Thanks for the suggestion!